internal/server/remove_team_member.go

```diff
@@ -0,0 +1,101 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/ministryofjustice/opg-sirius-user-management/internal/sirius"
+)
+
+type RemoveTeamMemberClient interface {
+	Team(context.Context, []*http.Cookie, int) (sirius.Team, error)
+	EditTeam(context.Context, []*http.Cookie, sirius.Team) error
+}
+
+type removeTeamMemberVars struct {
+	Path      string
+	SiriusURL string
+	Team      sirius.Team
+	Selected  []sirius.TeamMember
+	Errors    sirius.ValidationErrors
+}
+
+func removeTeamMember(client RemoveTeamMemberClient, tmpl Template, siriusURL string) Handler {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		if r.Method != http.MethodPost {
+			return StatusError(http.StatusMethodNotAllowed)
+		}
+
+		id, err := strconv.Atoi(strings.TrimPrefix(r.URL.Path, "/teams/remove-member/"))
+		if err != nil {
+			return StatusError(http.StatusNotFound)
+		}
+
+		team, err := client.Team(r.Context(), r.Cookies(), id)
+		if err != nil {
+			return err
+		}
+
+		vars := removeTeamMemberVars{
+			Path:      r.URL.Path,
+			SiriusURL: siriusURL,
+			Team:      team,
+		}
+
+		err = r.ParseForm()
+		if err != nil {
+			return StatusError(http.StatusBadRequest)
+		}
+
+		for _, id := range r.PostForm["selected[]"] {
+			userID, err := strconv.Atoi(id)
+			if err != nil {
+				return StatusError(http.StatusBadRequest)
+			}
+
+			for _, user := range team.Members {
+				if userID == user.ID {
+					vars.Selected = append(vars.Selected, user)
+				}
+			}
+		}
+
+		if r.PostFormValue("confirm") != "" {
+			var members []sirius.TeamMember
+			for _, member := range team.Members {
+				stillInTeam := true
+				for _, selected := range vars.Selected {
```

Comment:
You could probably clean up this nested loop if `vars.Selected` was a `map[int]sirius.TeamMember` (or even just `string` since you only use `DisplayName`?), not sure how much that would affect the template though

Reply:
Yep, switched to a map which makes the code much cleaner. Meant I couldn't to `{{ index .Selected 0 }}` to pull out the name in a sentence, but I just used `{{ range .Selected }}{{ . }}{{ end }}` since there's only one person in the list.